modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/object/validation/rule/CommerceReturnAccountEntryIdObjectValidationRuleEngineImpl.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.commerce.internal.object.validation.rule;
+
+import com.liferay.commerce.model.CommerceOrder;
+import com.liferay.commerce.service.CommerceOrderService;
+import com.liferay.object.constants.ObjectValidationRuleConstants;
+import com.liferay.object.scope.ObjectDefinitionScoped;
+import com.liferay.object.validation.rule.ObjectValidationRuleEngine;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.util.HashMapBuilder;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Crescenzo Rega
+ */
+@Component(service = ObjectValidationRuleEngine.class)
+public class CommerceReturnAccountEntryIdObjectValidationRuleEngineImpl
+	implements ObjectDefinitionScoped, ObjectValidationRuleEngine {
+
+	public static final String ACCOUNT_ENTRY_ID_VALIDATION_RULE_ENGINE =
+		"commerce-return-account-entry-id-validation-rule-engine";
+
+	@Override
+	public Map<String, Object> execute(
+		Map<String, Object> inputObjects, String script) {
+
+		Map<String, Object> results = HashMapBuilder.<String, Object>put(
+			"validationCriteriaMet", true
```

Comment:
isn't it better to start with a "false" statement and change it to true only at line 73 where it's the only case where we have a success?

Reply:
Hey @luca-pellizzon I followed the same approach as other validators already present

- [UniqueCompositeKeyObjectValidationRuleEngineImpl](https://github.com/CrescenzoRegaLF/liferay-portal/blob/LPD-20282/modules/apps/object/object-service/src/main/java/com/liferay/object/internal/validation/rule/UniqueCompositeKeyObjectValidationRuleEngineImpl.java)

- [DDMObjectValidationRuleEngineImpl](https://github.com/CrescenzoRegaLF/liferay-portal/blob/LPD-20282/modules/apps/object/object-service/src/main/java/com/liferay/object/internal/validation/rule/DDMObjectValidationRuleEngineImpl.java)